exercises/guillaume_communie/ex02_oo_basics/inc/ShapeSorter.h

```diff
@@ -0,0 +1,44 @@
+#ifndef SHAPE_SORTER_H
+#define SHAPE_SORTER_H
+
+#include "Shape.h"
+
+#include <vector>
+#include <string>
+
+/**
+ * A shape sorter
+ */
+class ShapeSorter {
+public:
+    /**
+     * Print shape that match a certain type
+     * @param[in] v The vector of shapes
+     * @param[in] t The type to match
+     */
+    static void printMatchType(std::vector<Shape*> v, std::string t);
```

Comment:
Composite object is better to pass by reference to avoid copy. If they are not going to be modified, const& is even more explicit.

Reply:
I changed to `const &` for the first two methods. I kept the copy for the others (see below).